### ISSUE TITLE
clang-tidy bugprone-use-after-move gives false positive warnings in the move constructor for derived classes

### ISSUE THREAD

**Issue by yabinc** [llvm/llvm-project#63202]
After https://github.com/llvm/llvm-project/commit/8c10256734cd47274671fcabe94f24f15ecd6209, bugprone-use-after-move checks constructor initialization list. But it gives a false positive warning in the move constructor of a derived class, when the constructor calls move constructor for a base class and moves some fields in the derived class.

An example is in https://cplusplus.com/forum/beginner/187808/. The code works. But clang-tidy gives below warning:

warning: 'sp' used after it was moved [bugprone-use-after-move]
   22 |     SpecialPerson(SpecialPerson&& sp) : Person{std::move(sp)}, surname{std::move(sp.surname)}{}
      |                                                                                  ^

To work around this, I can only use static_cast<Person&&>(sp) instead of std::move(sp).
Since this seems to be a common pattern of implementing move constructors for derived classes, it might worth fixing bugprone-use-after-move check in clang-tidy.

**Comment by sfc-gh-sgiesecke** [comment]
Also, this seems to be a regression in clang 17 vs. clang 16.

**Comment by sfc-gh-sgiesecke** [comment]
And the cause of this is probably 8c10256734cd47274671fcabe94f24f15ecd6209. While that change probably makes sense in other constructors, it should not apply in move constructors.

**Comment by T-Maxxx** [comment]
Well I get same error but with a bit different case (both clang and clang-tidy are from 20.1.4 release)
```cpp
SpecialPerson(SpecialPerson&& sp) noexcept : Person{std::forward<Person>(sp)} {
    // Read access to SpecialPerson fields through sp triggers bugprone-use-after-move.
}

SpecialPerson(SpecialPerson&& sp) noexcept : Person{static_cast<Person&&>(sp)} {
    // Read access to SpecialPerson fields through sp are OK.
}
```
Consider following std::forward implementation:
```cpp
template <class _Tp>
[[__nodiscard__]] inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR _Tp&&
forward(_LIBCPP_LIFETIMEBOUND __libcpp_remove_reference_t<_Tp>&& __t) _NOEXCEPT {
  static_assert(!is_lvalue_reference<_Tp>::value, "cannot forward an rvalue as an lvalue");
  return static_cast<_Tp&&>(__t);
}
```
